datalad/tests/test_config.py

```diff
@@ -703,6 +704,7 @@ def test_write_config_section(path=None):
     # can we handle a bare repo?
     gr = GitRepo(path, create=True, bare=True)
 
+    obscure = "ds-;&%b5{}'ΔЙקم๗あ.datc"
```

Comment:
might not be supported by the filesystem/locale etc. That is partially why we still carry https://github.com/datalad/datalad/blob/debian/debian/patches/deb_no_utf8 to patch away all such explicit uses. Please use `get_most_obscure_supported_name` or a `OBSCURE_FILENAME` const within desired obscure name.  Also if you use OBSCURE_FILENAME, like `f"ds-;#{OBSCURE_FILENAME}.datc` you likely be able to avoid needing ad-hoc `if on_windows:` condition below.  After all this fix is not about unicode but about `;` and `#` comments.

Reply:
Makes sense, will do! Thx!

Reply:
Done